### PR TITLE
Add mlx90614 sensors

### PR DESCRIPTION
**Pull request by jesserockz**
# What does this implement/fix?

Quick description and explanation of changes

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2845

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/mlx90614/mlx90614.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  buf[0] = this->address_;
+  buf[1] = reg;
+  buf[2] = data & 0xFF;
+  buf[3] = data >> 8;
+  uint8_t pec = this->crc8_pec_(buf, 4);
+
+  buf[0] = buf[2];
+  buf[1] = buf[3];
+  buf[2] = pec;
+  return this->write_bytes(reg, buf, 3);
```
Comment (GregJArnold) [inline esphome/components/mlx90614/mlx90614.cpp]: You could save a couple memory writes here.
```suggestion
  uint8_t buf[5];
  buf[0] = this->address_;
  buf[1] = reg;
  buf[2] = data & 0xFF;
  buf[3] = data >> 8;
  buf[4] = this->crc8_pec_(buf, 4);
  return this->write_bytes(reg, buf + 2, 3);
```

Comment (jesserockz) [reply esphome/components/mlx90614/mlx90614.cpp]: Haha, didnt think of that. Nice Thanks

**Comment by PsySc0rpi0n** [comment]
Just for the record
I have been following @jesserockz instructions to test this component for this sensor.
I just compile the last (I guess not anymore as I see things changing here on the fly) but this is the log I got with some improvements. Only Ambient Temp is still odd.
https://paste.debian.net/1252218/

**Comment by ghost** [comment]
Made a PR for the docs, and have added the test case

Which is easiest?
 - grab from https://github.com/notsonominal/esphome/commit/f1795c9dc3589e7b96afea1d97b109d16ecbb463
 - do a PR against jesserockz-2022-239 with just the test
 - do a PR against dev with the sensor component and test

**Pull request merged**